Added interface with OpenWebUI

TanguyBod opened the pull request:
Add support for OpenWebUI as UI option for RAG

- OpenWebUI can now be used as an alternative user interface.
- Ensured compatibility and responsiveness with existing RAG backend processes.

OpenWebUi must be launched separately and linked to colette in the connections section of the settings.
The built-in UI can still be used

Additional change : the sort_and_select_top_k function has been bypassed because it was causing very significant slowdowns in response times (About 2 minutes to answer a query on a 290-page document database). An alternative function has been put instead that just take the top-k documents. Time to answer is now around 10 seconds and quality of documents retrieved unchanged.

beniz commented:
Great, thanks! Can you update the docs so that there's an example that can be run ?

Also :
- we don't want custom prompts in .py files as much as possible, especially in the history, why do you need a prompt in hfmodel.py exactly ?
- don't comment ou the `filter` function, it's slow because of duplicate removal, instead make an option to choose yours or the original one, as far as I understand it / @fantes will check as well
- if the pure JSON API is not used along OpenWebUI, consider doing a httpopenweui.py file dedicated to that interface, along with the dedicated router.

TanguyBod commented:
- Docs are updated, I've added a page for integration with OpenWebUI.

- Custom prompt shouldn't be here, it was testing on my side, but the goal was to lead answers in a certain direction. Just out of curiosity, what's the reason(s) to avoid system prompts?

- Regarding the filter function, I've added a choice using the ad.parameters.input.rag.remove_duplicates boolean.

- The JSON API is still used to add or delete services to the server, some functions aren't used anymore, but I think it is preferable than duplicating half the httpjsonapi.py file.

beniz commented:
Great, please derive a `openwebuiapi.py` file, you don't need to duplicate functions, but separating the code is important for maintenance afterwards.
